app/src/main/java/com/android/unio/ui/association/EditAssociation.kt

```diff
@@ -0,0 +1,221 @@
+    Text(
+        text = context.getString(R.string.association_not_found),
+        color = Color.Red,
+        modifier = Modifier.padding(16.dp).testTag("associationNotFound"))
+  }
+}
+
+@Composable
+fun EditAssociationScaffold(
+    association: Association,
+    navigationAction: NavigationAction,
+    onSave: (Association) -> Unit
+) {
+  val context = LocalContext.current
+  val scope = rememberCoroutineScope()
+  var url by remember { mutableStateOf(TextFieldValue(association.url)) }
+  var name by remember { mutableStateOf(TextFieldValue(association.name)) }
+  var fullName by remember { mutableStateOf(TextFieldValue(association.fullName)) }
+  var description by remember { mutableStateOf(TextFieldValue(association.description)) }
+  var image by remember { mutableStateOf(TextFieldValue(association.image)) }
+
+  var expanded by remember { mutableStateOf(false) }
+  var category by remember { mutableStateOf(association.category) }
+
+  Column(modifier = Modifier.fillMaxSize().padding(16.dp).verticalScroll(rememberScrollState())) {
+    Text(
+        text = context.getString(R.string.edit_association_title),
+        style = MaterialTheme.typography.headlineMedium,
+        modifier = Modifier.testTag("EditAssociationTitle"))
+
+    Spacer(modifier = Modifier.height(16.dp))
+
+    // Explanation for "Name"
+    Text(
+        text = context.getString(R.string.name_explanation),
+        style = MaterialTheme.typography.bodySmall,
+        modifier = Modifier.testTag("NameExplanationText"))
+
+    Spacer(modifier = Modifier.height(8.dp))
+    OutlinedTextField(
+        value = name,
+        onValueChange = { name = it },
+        label = { Text("Name") },
+        modifier = Modifier.fillMaxWidth().testTag("NameTextField"))
+
+    Spacer(modifier = Modifier.height(16.dp))
+
+    // Explanation for "Full Name"
+    Text(
+        text = context.getString(R.string.full_name_explanation),
+        style = MaterialTheme.typography.bodySmall,
+        modifier = Modifier.testTag("FullNameExplanationText"))
+
+    Spacer(modifier = Modifier.height(8.dp))
+    OutlinedTextField(
+        value = fullName,
+        onValueChange = { fullName = it },
+        label = { Text("Full Name") },
+        modifier = Modifier.fillMaxWidth().testTag("FullNameTextField"))
+
+    Spacer(modifier = Modifier.height(16.dp))
+
+    // Explanation for "Category"
+    Text(
+        text = context.getString(R.string.category_explanation),
+        style = MaterialTheme.typography.bodySmall,
+        modifier = Modifier.testTag("CategoryExplanationText"))
+
+    Spacer(modifier = Modifier.height(8.dp))
+
+    // Category Button
+    Button(
+        onClick = { expanded = true },
+        modifier = Modifier.fillMaxWidth().testTag("CategoryButton")) {
+          Text(text = category.displayName)
+        }
+
+    DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
+      AssociationCategory.values().forEach { categoryOption ->
```

Comment:
Consider using `.entries` instead of `.values()`.

Reply:
you are correct this was depreciated